tutorials/time-freq/plot_ssvep.py

```diff
@@ -0,0 +1,786 @@
+# In these plots you can see that a shorter Welch window (relative to the
+# amount of data) leads not only to lower SNR values (as described above)
+# but also to lower variance / higher confidence in the means.
+#
+# This is the very idea of Welch's method, and is relevant to remember when
+# you decide for your analysis parameters:
+#
+# there is no such thing as an optimal welch window, its always a tradeoff:
+# a longer Welch window results in higher average SNR values;
+# a shorter window results in lower values but is more robust /
+# there is a higher confidence in the mean.
+#
+# Yet, one should also not pick too small a window: in our data we can see
+# that the very short Welch windows < 2-3s are not great -
+# here we've hit the noise bottom.
+#
+
+##############################################################################
+# Time resolved SNR
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# ..and finally we can trick MNE's PSD implementation to make it a
+# sliding window analysis and come up with a time resolved SNR measure.
+# This will reveal whether a participant blinked or scratched their head..
+#
+# Each of the ten trials is coded with a different color in the plot below.
+#
+
+# 3s sliding window
+window_length = 3
+window_starts = [i for i in range(20 - window_length)]
+window_snrs = [[]] * len(window_starts)
+
+for i_win, win in enumerate(window_starts):
+    # compute spectrogram
+    windowed_psd, windowed_freqs = mne.time_frequency.psd_welch(
+        epochs[str(event_id['12hz'])],
+        n_fft=int(sf * window_length) - 1, n_overlap=0, n_per_seg=None,
+        tmin=win, tmax=win + window_length,
+        fmin=fmin, fmax=fmax, verbose=False)
+    # define a bandwidth of 1hz around stimfreq for SNR computation
+    bin_width = windowed_freqs[1] - windowed_freqs[0]
+    noise_skip_neighborfreqs = \
+        round((stim_bandwidth / 2) / bin_width - bin_width / 2. - .5) if (
+            bin_width < stim_bandwidth) else 0
+    noise_n_neighborfreqs = \
+        int((sum((windowed_freqs <= 13) & (windowed_freqs >= 11)
+                 ) - 1 - 2 * noise_skip_neighborfreqs) / 2)
+    # compute snr
+    windowed_snrs = snr_spectrum(
+        windowed_psd,
+        noise_n_neighborfreqs=noise_n_neighborfreqs if (
+            noise_n_neighborfreqs > 0) else 1,
+        noise_skip_neighborfreqs=noise_skip_neighborfreqs)
+    window_snrs[i_win] = \
+        windowed_snrs[:, picks_roi_vis, np.argmin(
+            abs(np.subtract(windowed_freqs, 12.)))].mean(axis=1)
+
+plt.figure()
+plt.plot(window_starts, np.array(window_snrs))
+plt.title('Time resolved 12hz SNR - %is sliding window' % window_length)
+plt.legend(['individual trials'])
+plt.ylabel('average SNR')
+plt.xlabel('t0 of analysis window [s]')
+plt.axhline(1, ls='--', c='r')
+plt.show()
+
+##############################################################################
+# Well.. turns out this was a bit too optimistic ;)
```

Comment:
The runtime on CircleCI is over a minute. Ideally an example like this would be < 30 sec. Do you see places where it could be cut down a bit?